samza-core/src/main/java/org/apache/samza/zk/ZkBarrierForVersionUpgrade.java

```diff
@@ -31,6 +33,8 @@
   private final ZkUtils zkUtils;
```

Comment:
Can we make this ZkBarrier general like ZkLeaderElector? LocalApplicationRunner also needs this to be able to have the barrier for environment initialization (e.g creating streams). By looking at the code it seems if we replace "version" with a general "name" then this can be a general Barrier. Can we make this change so this class can be reused by LocalApplicationRunner?